### PR TITLE
Add a "StreamingHash" class that's externally instantiatable 

### PR DESCRIPTION
**Pull request by justinwsmith**
Addresses issue: https://github.com/nashby/xxhash/issues/9

### CODE REVIEW

**Comment by justinwsmith** [comment]
I made additional changes:
* I moved the external classes to the Digest module.  The are now available Digest::XXHash, Digest::XXHash32 and Digest::XXHash64 classes. These classes are intended to create objects that behave like the other digests in this module.
* I added methods "reset" and "digest!" to increase its compatibility with other digests.

File: ext/xxhash/xxhash.c [hunk 1]
```diff
@@ -1,33 +1,55 @@
 #include "xxhash.h"
 
+
+
```
Comment (nashby) [inline ext/xxhash/xxhash.c]: can you please remove extra spaces here?

File: lib/xxhash.rb [hunk 2]
```diff
@@ -33,4 +34,58 @@ def self.xxh64_stream(io, seed = 0, chunk_size = 32)
 
     hash.digest
   end
+
+end
+
+module Digest
+  class XXHash < Digest::Class
+    attr_reader :digest_length
+    def initialize bitlen, seed = 0
```
Comment (nashby) [inline lib/xxhash.rb]: can you please use parentheses here and everywhere below? Thanks!

Comment (justinwsmith) [inline lib/xxhash.rb]: It's not clear the scope of this request.  Use of parentheses for method
calls is generally not idiomatic in Ruby.

Comment (justinwsmith) [inline lib/xxhash.rb]: OK. I'll take a closer look later.

**Comment by tilllt** [comment]
Is this going to be merged at some point, as far as i understand only ruby-xxhash can be instantiated through Digest::, but not the xxhash wrapper... Why?

Or is it done and i misunderstood?

**Comment by justinwsmith** [comment]
I think it's complete and tested. I'm not sure why that pull request has
not yet been merged.

**Comment by nashby** [comment]
@justinwsmith thank you for the pull request and sorry it took so long to merge it!

**Comment by tilllt** [comment]
Great, thanks to both of you!

**Comment by tilllt** [comment]
Hmm, now i get this: 
hashsum = Digest::XXHash32.new

results in

ruby-2.1.5/lib/ruby/2.1.0/digest.rb:12:in `const_missing': library not found for class Digest::XXHash32 -- digest/xxhash32 (LoadError)

totally possible though i am doing something wrong...

**Comment by justinwsmith** [comment]
That looks like a runtime linking error - it couldn't find the xxhash library. I've seen that error, but I don't recall the best fix for it. Did you build the gem yourself?

Sent from Outlook Mobile




On Wed, Feb 10, 2016 at 4:04 AM -0800, "tilllt" <notifications@github.com> wrote:












Hmm, now i get this: 

hashsum = Digest::XXHash32.new



results in



ruby-2.1.5/lib/ruby/2.1.0/digest.rb:12:in `const_missing': library not found for class Digest::XXHash32 -- digest/xxhash32 (LoadError)



totally possible though i am doing something wrong...

**Comment by tilllt** [comment]
no, i have this in my gemfile:
gem 'xxhash', :git => 'git://github.com/nashby/xxhash.git'

**Pull request merged**